src/sys/unix/tcp.rs

```diff
@@ -438,12 +459,36 @@ pub fn accept(listener: &net::TcpListener) -> io::Result<(net::TcpStream, Socket
         target_os = "openbsd"
     ))]
     let stream = {
-        syscall!(accept4(
-            listener.as_raw_fd(),
-            addr.as_mut_ptr() as *mut _,
-            &mut length,
-            libc::SOCK_CLOEXEC | libc::SOCK_NONBLOCK,
-        ))
+        use std::sync::atomic::{AtomicBool, Ordering};
+
+        static ACCEPT4_AVAIL: AtomicBool = AtomicBool::new(true);
+
+        if ACCEPT4_AVAIL.load(Ordering::Relaxed) {
+            let result = syscall!(accept4(
+                listener.as_raw_fd(),
+                addr.as_mut_ptr() as *mut _,
+                &mut length,
+                libc::SOCK_CLOEXEC | libc::SOCK_NONBLOCK,
+            ));
+
+            match result {
+                Ok(s) => Ok(s),
+                Err(err) => {
+                    match err.raw_os_error() {
+                        Some(libc::ENOSYS) => {
```

Comment:
`EPERM` should probably be added to as a fallback condition here, because if `accept4` is blocked by seccomp, that is going to be returned.

Rust std usually does it like this when they use newer syscalls.

Reply:
Ok.